### PR TITLE
Roylly new arrow orientation

### PR DESCRIPTION
**Pull request by roylly**
### Checklist for this pull request

- [x] I have read the [guidelines for contributing](../CONTRIBUTING.md).
- [x] I have updated README.md, describing the new feature I'm adding (if applicable).
- [x] I have checked that my code additions did fail neither code linting checks nor unit/UI test.

### Description

I have finished the arrow orientation support left and right development, but don't make big change for the layout engine, just add new case for layout.

Thanks!

### CODE REVIEW

**Comment by ephread** [comment]
Thanks for taking the time to rework your PR! I haven't forgotten you, I'll take a look before the end of the coming week.

**Comment by ephread** [comment]
Thanks for your work, it's much cleaner now, which make my job easier. I really appreciate it!

I see two issues remaining:

1. You need to install [Swiftlint](https://github.com/realm/SwiftLint) and make sure the code pass linting. I don't have a strict style guide, but passing the linting phase with neither warnings nor errors is required.

2. You went for manual positioning, which means that left/right settings override the automated positioning. Nothing wrong with that, I'm happy with it. However, I don't think we should be forced to provide a `maxWidth` to prevent coach marks from being out of bounds. The second coach mark in the provided example is partially out of bounds on an iPhone SE, while it should only expand as far as the screen bounds (if `maxWidth` is not provided).

Thanks for being patient and reworking your PR. We're getting there!

File: Sources/Instructions/Core/Internal/CoachMarksViewController.swift [hunk 1]
```diff
@@ -81,7 +81,7 @@ class CoachMarksViewController: UIViewController {
     var customStatusBarStyle: UIStatusBarStyle?
 
     ///
-    override var preferredStatusBarStyle: UIStatusBarStyle {
+    override public var preferredStatusBarStyle: UIStatusBarStyle {
```
Comment (ephread) [inline Sources/Instructions/Core/Internal/CoachMarksViewController.swift]: Needs to remain `internal` as well. 🙂

Comment (roylly) [reply Sources/Instructions/Core/Internal/CoachMarksViewController.swift]: sorry ,forget revert that.

**Comment by roylly** [comment]
I have fixed the two issues you pointed out . thanks!  And i am learning swift, thanks for your help.

**Comment by ephread** [comment]
Looks good to me!

> And i am learning swift, thanks for your help.

You're welcome ;)

**Comment by bryan1anderson** [comment]
Nearly a year old. Has this not been merged in? I'm only seeing "top, bottom"

**Comment by ephread** [comment]
@bryan1anderson Yeah, sorry about that. This was merged into a temporary branch. There's still a bit of cleanup before I can make it play well with `master` and it's low priority for me.

**Pull request merged**